src/PerimeterxContext.php

```diff
@@ -399,6 +400,15 @@ public function setDecodedCookie($cookie)
         $this->decoded_px_cookie = $cookie;
     }
 
+
+    private function isSensitiveRoutePrefix($sensetive_routes_prefix, $uri){
+      foreach($sensetive_routes_prefix as $route){
+        if (strpos($uri, $route) === 0){
```

Comment:
consider using strncmp with the route length - in this case we will not scan the full uri to know that the prefix is not there.